File: BookingApp/src/app/RentACar/Services/BranchService/branch.service.ts
import { Injectable } from '@angular/core';
import { AddBranchParameters } from 'src/app/Shared/Model/RentACars/Models/Parameters/AddBranchParameters.model';
import { EditBranchParameters } from 'src/app/Shared/Model/RentACars/Models/Parameters/EditBranchParameters.model';
import { HttpClient, HttpParams } from '@angular/common/http';


@Injectable({
  providedIn: 'root'
})
export class BranchService {

  constructor(private client : HttpClient) { }

  //metoda za dodavanje nove filijale, koju poziva rentACarAdmin odredjene kompanije
  addBranch(enterpriseBranch: AddBranchParameters){
    return this.client.post('http://localhost:50000/api/Branch/AddBranch', enterpriseBranch);
  }

  //metoda za izmenu jedne filijale, koju poziva rentACarAdmin odredjene kompanije
  editBranch(enterpriseBranch: EditBranchParameters){
    return this.client.put('http://localhost:50000/api/Branch/EditBranch', enterpriseBranch);
  }

  //metoda za brisanje jedne filijale, koju poziva rentACarAdmin odredjene kompanije
  deleteBranch(enterpriseId: number, branchId: number){
    return this.client.delete('http://localhost:50000/api/Branch/DeleteBranch',{
      params: new HttpParams().set('enterpriseId', enterpriseId.toString()).set('branchId', branchId.toString())
    })
  }

  //metoda za pribavljanje svih filijala odredjene rentACar kompanije
  getAllBranches(enterpriseId: number){
    return this.client.get('http://localhost:50000/api/Branch/GetAllBranches', {
      params: new HttpParams().set('enterpriseId', enterpriseId.toString())
    });
  }

  //metoda koja pribavlja informacije o jednoj filijali odredjene kompanije
  getOneBranch(enterpriseId: number, branchId: number){
    return this.client.get('http://localhost:50000/api/Branch/GetOneBranch', {
      params: new HttpParams().set('enterpriseId', enterpriseId.toString()).set('branchId', branchId.toString())
    });
  }

  //metoda koja pribavlja informacije o adresi jedne filijale 
  getBranchLocationOnMap(branchId: number){
    return this.client.get('http://localhost:50000/api/Branch/GetBranchAddress', {
      params: new HttpParams().set('branchId', branchId.toString())
    });
  }
}

File: BookingAppBackend/BookingAppBackend/Service/RentACar/Branch/BranchService.cs
﻿using BookingAppBackend.Database.Interfaces;
using BookingAppBackend.Database.Interfaces.RentACar;
using BookingAppBackend.Model.RentACar;
using BookingAppBackend.Model.RentACar.Parameters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BookingAppBackend.Service.RentACar.Branch
{
    public class BranchService : IBranchService
    {
        private IEnterpriseBranchRepository repo;
        private IUnitOfWork unitOfWork;

        public BranchService(IEnterpriseBranchRepository repo, IUnitOfWork unitOfWork)
        {

            this.repo = repo;
            this.unitOfWork = unitOfWork;
        }

        public async Task<EnterpriseBranch> AddBranch(AddBranchParameters enterpriseBranch)
        {
            try
            {
                var temp = await repo.AddBranch(enterpriseBranch);
                if (temp != null)
                {
                    await unitOfWork.CompleteAsync();
                }
                return temp;
            }
            catch
            {
                return null;
            }
        }

        public async Task<EnterpriseBranch> DeleteBranch(int enterpriseId, int branchId)
        {
            try
            {
                var temp = await repo.DeleteBranch(enterpriseId, branchId);
                if (temp != null)
                {
                    await unitOfWork.CompleteAsync();
                }
                return temp;
            }
            catch
            {
                return null;
            }
        }

        public async Task<EnterpriseBranch> EditBranch(EditBranchParameters enterpriseBranch)
        {
            try
            {
                var temp = await repo.EditBranch(enterpriseBranch);
                if (temp != null)
                {
                    await unitOfWork.CompleteAsync();
                }
                return temp;
            }
            catch
            {
                return null;
            }
        }

        public async Task<IEnumerable<EnterpriseBranch>> GetAllBranches(int enterpriseId)
        {
            try
            {
                var temp = await repo.GetAllBranches( enterpriseId);
                
                return temp;
            }
            catch
            {
                return null;
            }
        }

        public async Task<EnterpriseBranch> GetBranchAddress(int branchId)
        {
            try
            {
                var temp = await repo.GetBranchAddress(branchId);

                return temp;
            }
            catch
            {
                return null;
            }
        }

        public async Task<EnterpriseBranch> GetOneBranch(int enterpriseId, int branchId)
        {
            try
            {
                var temp = await repo.GetOneBranch(enterpriseId, branchId);
                
                return temp;
            }
            catch
            {
                return null;
            }
        }
    }
}
